File: week2/게임_맵_최단거리/황유연_게임_맵_최단거리.java
import java.util.ArrayDeque;
import java.util.Queue;

class Solution {
    public int solution(int[][] maps) {
        int answer = 0;
        int n = maps.length;
        int m = maps[0].length;
        int[] dx = {0, 0, -1, 1};
        int[] dy = {1, -1, 0, 0};
        Queue<int[]> q = new ArrayDeque<>();
        q.add(new int[]{0, 0, 1});

        int[][] visited = new int[n][m];
        visited[0][0] = 1;

        while (!q.isEmpty()) {
            int[] curr = q.poll();
            int y = curr[0];
            int x = curr[1];
            int d = curr[2];

            if (y == n - 1 && x == m - 1) {
                return d;
            }
            for (int i = 0; i < 4; i++) {
                int ny = y + dy[i];
                int nx = x + dx[i];
                if (0 <= ny && ny < n && 0 <= nx && nx < m && visited[ny][nx] != 1 && maps[ny][nx] == 1) {
                    visited[ny][nx] = 1;
                    q.add(new int[]{ny, nx, d + 1});
                }
            }

        }

        return -1;
    }


}

File: week2/게임_맵_최단거리/황유연_게임_맵_최단거리.py
from collections import deque


def solution(maps):
    n = len(maps)
    m = len(maps[0])

    def bfs(y, x):
        q = deque()
        q.append((y, x, 1)) # y좌표, x좌표, 거리
        visited = [[0] * m for _ in range(n)] # 방문 표시
        visited[y][x] = 1
        dy, dx = [0, 0, 1, -1], [1, -1, 0, 0]

        while q:
            y, x, d = q.popleft()
            if y == n - 1 and x == m - 1:
                return d
            for i in range(4):
                ny = y + dy[i]
                nx = x + dx[i]
                if 0 <= ny < n and 0 <= nx < m and not visited[ny][nx] and maps[ny][nx] == 1: # 범위 내, 방문x, 장애물없는지 체크
                    visited[ny][nx] = 1
                    q.append((ny, nx, d + 1))
        return -1

    return bfs(0, 0)

